verilog/analysis/checkers/struct_union_name_style_rule.cc

```diff
@@ -81,13 +83,38 @@ void StructUnionNameStyleRule::HandleSymbol(const verible::Symbol& symbol,
     }
     const auto* identifier_leaf = GetIdentifierFromTypeDeclaration(symbol);
     const auto name = ABSL_DIE_IF_NULL(identifier_leaf)->get().text();
-    if (!verible::IsLowerSnakeCaseWithDigits(name) ||
+    // get rid of the exceptions specified by user
+    std::string name_filtered;
+    if (exceptions_.size()) {
+      name_filtered.reserve(name.length());
+      const auto& name_split = absl::StrSplit(name, '_');
+      for (const auto& ns : name_split) {
+        if (std::find(exceptions_.begin(), exceptions_.end(), ns) ==
+            exceptions_.end()) {
+          name_filtered.append(ns);
+        }
+      }
+    }
```

Comment:
After this, a string `foo_bar_baz` will look like `foobarbaz` as we also removed the underscores.

In particular strings that previously are not allowed, a string with a starting underscore `_foo_bar` will now be `foobar` and becomes allowed.

So I think adding a test-case for this scenario should help figuring out an implementation that covers that.